src/renderer/components/database-diagram-modal.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component, PropTypes } from 'react';
+import DatabaseDiagram from './database-diagram.jsx';
+import Loader from './loader.jsx';
+
+const STYLE = {
+  list: { maxHeight: '250px', overflow: 'hidden', overflowY: 'scroll', padding: '8px', border: '2px solid' },
+};
+
+
+export default class DatabaseDiagramModal extends Component {
+  static propTypes = {
+    database: PropTypes.string,
+    tables: PropTypes.array,
+    selectedTables: PropTypes.array,
+    views: PropTypes.array,
+    columnsByTable: PropTypes.object,
+    references: PropTypes.object,
+    onShowDatabaseDiagram: PropTypes.func.isRequired,
+    onClose: PropTypes.func.isRequired,
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    $(this.refs.diagramModal).modal({
+      closable: true,
+      detachable: false,
+      // Updates modal position on loading diagram in modal DOM
+      observeChanges: true,
+      onHidden: () => {
+        this.props.onClose();
+      },
+    }).modal('show');
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.showDiagramIfNeeded(nextProps);
+  }
+
+  onSelectAllTables() {
+    $(':checkbox', 'div.ui.list').prop('checked', true);
```

Comment:
Do we really need JQuery to set values for the checkbox? Why not use only react for that?